coco/scheduler.py

```diff
@@ -114,16 +125,65 @@ def _wait_time(self):
 
 
 class EndpointTimer(Timer):
-    """Timer that calls a coco endopint."""
+    """Timer that calls a coco endpoint."""
 
     def __init__(self, period, endpoint, host, port):
         self.endpoint = endpoint
         self.host, self.port = host, port
+        self._check = []
         super().__init__(endpoint.name, period)
 
+    def add_condition(self, condition):
+        """
+        Add a condition on the state that must be satisfied before calling the scheduled endpoint.
+
+        Parameters
+        ----------
+        condition: dict
+            Must include at least key 'path' indicating state field to check.
+            If 'value' and 'type' are included, ensure the state has this value,
+            otherwise just check it exists.
+        """
+        try:
+            path = condition["path"]
+        except KeyError:
+            raise KeyError(f"Endpoint '{self.name}' conditions must include field 'path'.")
+        check = {"path": path}
+        val = condition.get("value", None)
+        if val is not None:
+            try:
+                val_type = locate(condition["type"])
+            except KeyError:
+                raise KeyError(
+                    f"Endpoint '{self.name}' conditions must include field 'type'"
+                    " when 'value' is specified."
+                )
+            if val_type is None:
+                raise ValueError(f"'require_state' of endpoint {self.name} is of unknown type.")
+            check.update({"value": val_type(val), "type": val_type})
+        self._check.append(check)
+
     async def _call(self):
         # logger.debug(f"{self.name}: {time() - self._start_t}")
 
+        # Check conditions are satisfied
+        for c in self._check:
+            try:
+                state_val = self.endpoint.state.read(c["path"])
+            except KeyError:
+                logger.info(
+                    f"Skipping scheduled endpoint {self.name} because {c['path']} doesn't exist."
+                )
+                return
+            val = c.get("value", None)
+            if val is not None:
+                logger.debug(f"{val}, state {state_val}")
+                state_val = c["type"](state_val)
```

Comment:
I don't get this. Is the `type` in the config not for an optional type check?

Reply:
Well I made it such that if val/type are omitted it just checks if that path exists. We could have it type check, I don't have an opinion as to what is preferable.